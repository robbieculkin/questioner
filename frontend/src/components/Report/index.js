import React, { Component } from 'react';
import axios from 'axios';

import './index.scss';

import Feed from '../Feed';

class Report extends Component {
  constructor(props) {
    super(props);

    this.state = {
      history: []
    };
  }

  componentDidMount() {
    axios.get('http://localhost:5000/api/v0/report')
      .then(res => {
          const data = res.data;

          this.setState({
            history: data.history
          })
      });
  }

  render() {
    const { history } = this.state;

    return (
      <div className='container centered'>
        <div className='title'>
          <h1>Discussion Report</h1>
        </div>
        <div className='tile'>
          <Feed history={history} />
        </div>
      </div>
    );
  }
}

export default Report;
