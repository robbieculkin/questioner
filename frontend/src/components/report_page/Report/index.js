import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import axios from 'axios';
import PropTypes from 'prop-types';

import './index.scss';

import Feed from '../Feed';

import { errorState, emptyState } from '../../../data/errorData';
import { REPORT_URI } from '../../../config/api';

class Report extends Component {
  constructor(props) {
    super(props);

    this.state = {
      sessionId: props.location.state.sessionId,
      selectedPlay: 'Loading...',
      history: []
    };
  }

  componentDidMount() {
    const { sessionId } = this.state;
    this._mounted = true;

    axios.get(REPORT_URI, { params: { sessionId } })
      .then(res => {
        if (!this._mounted)
          return;

        const { selectedPlay, discussion } = res.data;
        this.setState({ selectedPlay });
        if (discussion !== undefined && res.data.discussion.length > 0) {
          this.setState({  history: res.data.discussion });
        } else {
          this.setState(emptyState);
        }
      })
      .catch(error => {
        if (!this._mounted) return;
        console.log(error);
        this.setState(errorState);
      });
  }

  componentWillUnmount() {
    this._mounted = false;
  }

  render() {
    const { sessionId, selectedPlay, history } = this.state;

    return (
      <div className='container centered'>
        <div className='title'>
          <h1>Discussion Report</h1>
        </div>
        <div className='details'>
          <p>Session ID: {sessionId}</p>
          <p>Selected play: {selectedPlay}</p>
        </div>
        <div className='tile'>
          <Feed history={history} />
        </div>
        <Link to='/' className='link'>
          <div className='main-button tile'>Home</div>
        </Link>
      </div>
    );
  }
}

Report.propTypes = {
  location: PropTypes.shape({
    state: PropTypes.shape({
      sessionId: PropTypes.string
    })
  })
};

export default Report;
