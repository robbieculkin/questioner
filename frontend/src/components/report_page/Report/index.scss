.details {
  padding-left: 150px;
  p {
    font-size: 16px;
    margin-top: 5px;
    margin-bottom: 0;
  }
}