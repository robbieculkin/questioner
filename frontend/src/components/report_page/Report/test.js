import React from 'react';
import { shallow } from 'enzyme';
import Report from '.';

describe('Report', () => {
  it('should render correctly', () => {
    const reportProps = { location: { state: { sessionId: '123uuid' }}};
    const component = shallow(<Report {...reportProps} />);
    expect(component).toMatchSnapshot();
  });
});
