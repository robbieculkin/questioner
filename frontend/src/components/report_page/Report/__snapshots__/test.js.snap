// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Report should render correctly 1`] = `
<div
  className="container centered"
>
  <div
    className="title"
  >
    <h1>
      Discussion Report
    </h1>
  </div>
  <div
    className="details"
  >
    <p>
      Session ID: 
      123uuid
    </p>
    <p>
      Selected play: 
      Loading...
    </p>
  </div>
  <div
    className="tile"
  >
    <Feed
      history={Array []}
    />
  </div>
  <Link
    className="link"
    replace={false}
    to="/"
  >
    <div
      className="main-button tile"
    >
      Home
    </div>
  </Link>
</div>
`;
